lms/static/js/instructor_dashboard/membership.js

```diff
@@ -206,12 +212,74 @@ such that the value can be defined later than this assignment (file load order).
                             );
                         }
                     } else {
-                        authListWidgetReloadList.add_row([member.username, member.email, $revokeBtn]);
+
+                        var row = [member.username, member.email, $revokeBtn];
+                        if (authListWidgetReloadList.is_cohort_enable) {
+                            row = [
+                                ...row,
+                                authListWidgetReloadList.render_cohort_settings_control(member)
+                            ];
+                        }
+                        authListWidgetReloadList.add_row(row);
                     }
                 });
             });
         };
 
+        AuthListWidget.prototype.render_cohort_settings_control = function (member) {
+            var authListWidgetReloadList = this;
+            var assigment_info = member.cohorts_assignment;
+            let all_cohort_decorator = function (cohort_assignment) {
+                return [
+                    {
+                        id: -1,
+                        name: gettext('All'),
+                        is_assignment: cohort_assignment.every(info => info.is_assignment)
+                    },
+                    ...cohort_assignment
+                ]
+            };
+
+            var checkers_list = all_cohort_decorator(assigment_info).map(cohort_info => `
```

Comment:
Please, clarify why we generate HTML inside js code?

Reply:
As for me, an additional template will be unnecessarily